src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -52,41 +105,267 @@ class LogManager {
+    {
+      std::unique_lock<std::mutex> lock(persist_lock_);
+      disk_log_writer_thread_cv_.notify_one();
+    }
+    // Mark that serializer thread doesn't have a buffer in its possession to which it can write to
+    commits_in_buffer_.clear();
+    filled_buffer_ = nullptr;
+  }
+
+  /**
+   * If the central thread registry grants us a thread, we accept if we currently don't have a thread to run the disk
+   * consumer task. Else we don't need the thread, so we respectfully decline.
+   * @return true if we accepted the thread, else false
+   */
+  bool OnThreadOffered() override {
+    if (GetThreadCount() == 0) {
+      // Register disk log consumer task
+      TERRIER_ASSERT(disk_log_writer_task_ == nullptr, "We should not have a task if we don't own a thread for it yet");
+      disk_log_writer_task_ = DedicatedThreadRegistry::GetInstance().RegisterDedicatedThread<DiskLogConsumerTask>(
+          this /* requester */, this /* argument to task constructor */);
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * If the central registry wants to removes our thread used for the disk log consumer task, we only allow removal if
+   * we are in shut down, else we need to keep the task, so we reject the removal
+   * @return true if we allowed thread to be removed, else false
+   */
+  bool OnThreadRemoval(common::ManagedPointer<DedicatedThreadTask> task) override {
+    // We don't want to register a task if the log manager is shutting down though.
+    return !run_log_manager_;
+  }
+};
+
+/**
+ * Task that processes buffers handed over by transactions and serializes them into consumer buffers
+ */
+class LogSerializerTask : public DedicatedThreadTask {
+ public:
+  /**
+   * @param log_manager Pointer to log manager
+   * @param serialization_interval Interval time for when to trigger serialization
+   */
+  explicit LogSerializerTask(LogManager *log_manager, const std::chrono::milliseconds serialization_interval)
+      : log_manager_(log_manager), serialization_interval_(serialization_interval), run_task_(false) {}
+
+  /**
+   * Runs main disk log writer loop. Called by thread registry upon initialization of thread
+   */
+  void RunTask() override {
+    run_task_ = true;
+    LogSerializerTaskLoop();
+  }
+
+  /**
+   * Signals task to stop. Called by thread registry upon termination of thread
+   */
+  void Terminate() override { run_task_ = false; }
+
+ private:
+  LogManager *log_manager_;
+  const std::chrono::milliseconds serialization_interval_;
+  bool run_task_;
+
+  /**
+   * Main serialization loop. Calls Process on LogManager every interval
+   */
+  void LogSerializerTaskLoop() {
+    while (run_task_) {
+      std::this_thread::sleep_for(serialization_interval_);
+      log_manager_->Process();
+    }
   }
 };
+
+/**
+ * Task signals disk log consumer to flush logs into disk
+ */
+class LogFlusherTask : public DedicatedThreadTask {
```

Comment:
nit: If you break up LogManager logic into these two classes, they should get their own files.

Reply:
Done.